diff --git a/modules/contrib/lti_tool_provider/src/Authentication/Provider/LTIToolProviderBase.php b/modules/contrib/lti_tool_provider/src/Authentication/Provider/LTIToolProviderBase.php
index b6ea9c3e..f30a9286 100644
--- a/modules/contrib/lti_tool_provider/src/Authentication/Provider/LTIToolProviderBase.php
+++ b/modules/contrib/lti_tool_provider/src/Authentication/Provider/LTIToolProviderBase.php
@@ -135,6 +135,34 @@ abstract class LTIToolProviderBase implements AuthenticationProviderInterface {
     $name = $context->getUserIdentity()->getName();
     $mail = $context->getUserIdentity()->getEmail();
 
+    $version=$context->getVersion();
+    $full_context = [];
+    if ($version == 'V1P3') {
+      if (empty($name)) {
+        $name = $context->getUserIdentity()->getIdentifier();
+      }
+      if (empty($mail)) {
+        $mail = $name.'@lib.umich.edu';
+      }
+      $payload = $context->getPayload();
+      $claims = $payload->getToken()->getClaims()->all();
+      $full_context['lis_person_name_full'] = $claims['name'];
+      $roles = $claims['https://purl.imsglobal.org/spec/lti/claim/roles'];
+      if (in_array('http://purl.imsglobal.org/vocab/lis/v2/membership#Learner', $roles) &&
+          in_array('http://purl.imsglobal.org/vocab/lti/system/person#TestUser', $roles)) {
+        $full_context['roles'] = 'Learner';
+      }
+    }
+    else {
+      $full_context = $context->getContext();
+    }
+
+    if ((isset($full_context['lis_person_name_full']) && $full_context['lis_person_name_full'] == 'Test Student') &&
+       (isset($full_context['roles']) && $full_context['roles'] == 'Learner')) {
+      $name = 'Test Student';
+      $mail = 'teststudent@umich.edu';
+    }
+
     if (empty($name)) {
       throw new Exception('Name not available for user provisioning.');
     }
